### PR TITLE
feat(financial-statements-inao): Add application api template module

### PR DESCRIPTION
**Pull request by fjandakornid**
# Add application api template module for Financial Statements IANO

## What

- Added application api template module for Financial Statements INAO
- Moved client from api-module to it's own lib
- Changed connection for Dataverse from sandbox to RE environment

## Why

Needed to get current user's client type from Dataverse in applications x-state onEntry function.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/application-system/api/infra/application-system-api.ts
Comment (pshomov) [inline apps/application-system/api/infra/application-system-api.ts:78]: I cannot find where this flag is used in the code. Unconditional use of feature-specific environment variable can be dangerous. For more info on server-side features, please see - https://docs.devland.is/libs/nest/config#server-side-features

Comment (fjandakornid) [reply apps/application-system/api/infra/application-system-api.ts:78]: The flag isn't used in the code. This is because these variables doesn't exist in the AWS parameter store and this is to prevent them to be configured there. I was copying Snorri's work in this PR: https://github.com/island-is/island.is/pull/7685

**Review by obmagnusson — APPROVED** [review]
Application System files look good. Although you might want to update the codeowners file.

File: apps/application-system/api/infra/application-system-api.ts
Comment (pshomov) [reply apps/application-system/api/infra/application-system-api.ts:78]: That was the part where we configure the deployment and the feature. But The code needs to respect the flag and not throw because of missing env variables, which is what happened during the last release to prod.

**Comment by fjandakornid** [comment]
> @fjandakornid Is this PR "on hold"? I am not sure if I am supposed to review something here after our last conversion.

I just made the changes we talked about and it can be reviewed now. I will send you updated secret values before merging this.

**Review by saevarma — CHANGES_REQUESTED** [review]
Please rebase with main and review the upgrade guide https://www.notion.so/Nx-13-10-5-a77cd802539e468596da6265700ff3cf.

**Review by FridrikMProgramm — APPROVED** [review]

**Comment by andes-it** [comment]
This PR's had a 15-day siesta. Lost its spark, did it? It's making an exit in 5 days unless the stale label disappears.

**Comment by andes-it** [comment]
This PR closed its doors. Sat around for a day after the stale tag, then reckoned it's not worth the fuss.

**Pull request closed without merging**
